neqo-bin/src/client/http3.rs

```diff
@@ -447,4 +439,12 @@ impl UrlHandler {
         self.stream_handlers.remove(&stream_id);
         self.process_urls(client);
     }
+
+    fn reinit(&mut self) {
+        for url in self.handled_urls.drain(..).rev() {
```

Comment:
Copy from above, except the inserted `rev()` to maintain the URL queue order.

Reply:
Today I learned: `rev()` uses [`DoubleEndedIterator`](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html), which actually makes this fast. (Not implying speed is relevant here.)

Reply:
This only maintains *approximately* the same order.  Any URL that needed a retry will land in `handled_urls` out of order.  I guess "some effort" is better than "no effort", but I would have instead done:

```rust
self.url_queue.append(&mut self.handled_urls);
```

Though this would have required changing the type of the other.

Reply:
Leaving as is. Let me know in case you feel strongly about the above.